### DISCUSSION TITLE
Error when building with gcc 14.x due to uninitialized union in etherfabric.c

### DISCUSSION THREAD

**Discussion by forwardcomputers** [ipxe/ipxe#1284] [General]
Hi,

Trying to compile ipxe on Debian trixie 12.2, and keep getting this error.

Applying the change in this pull request [https://github.com/ipxe/ipxe/pull/1271](https://github.com/ipxe/ipxe/pull/1271), fixed the issue.

Ali.

    /ipxe/src# make -j 4
    In file included from drivers/net/etherfabric.c:36:
    In function 'falcon_xaui_link_ok',
        inlined from 'falcon_init_xmac' at drivers/net/etherfabric.c:2355:14,
        inlined from 'falcon_init_xmac' at drivers/net/etherfabric.c:2314:1:
    drivers/net/etherfabric.h:524:28: error: 'reg.u32[0]' may be used uninitialized [-Werror=maybe-uninitialized]
    522 |         (dword).u32[0] = ( ( (dword).u32[0]                                   \
        |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    523 |                              & ~EFAB_INPLACE_MASK32 (  0,  31, field ) )      \
        |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    524 |                            | EFAB_INSERT_FIELD32 (  0,  31, field, value ) ); \
        |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/net/etherfabric.c:2233:9: note: in expansion of macro 'EFAB_SET_DWORD_FIELD'
    2233 |         EFAB_SET_DWORD_FIELD ( reg, FCN_XX_DISPERR, FCN_XX_DISPERR_RESET);
        |         ^~~~~~~~~~~~~~~~~~~~
    drivers/net/etherfabric.c: In function 'falcon_init_xmac':
    drivers/net/etherfabric.c:2213:22: note: 'reg.u32[0]' was declared here
    2213 |         efab_dword_t reg;
        |                      ^~~
    cc1: all warnings being treated as errors
    make: *** [Makefile.housekeeping:964: bin/etherfabric.o] Error 1
    make: *** Waiting for unfinished jobs....

    /ipxe/src# uname -a
    Linux update 6.8.8-1-pve #1 SMP PREEMPT_DYNAMIC PMX 6.8.8-1 (2024-06-10T11:42Z) x86_64 GNU/Linux

    /ipxe/src# as --version
    GNU assembler (GNU Binutils for Debian) 2.43.1

    /ipxe/src# gcc --version
    gcc (Debian 14.2.0-1) 14.2.0

**Comment by NiKiZe** [comment]
Already open issue https://github.com/ipxe/ipxe/issues/1219
